amazons3.module

```diff
@@ -541,7 +541,12 @@ function amazons3_form_bucket_validate(array &$element, array &$form_state, arra
   // Inject our credentials for testing.
   $config = array();
   if (isset($form_state['values']['amazons3_key']) && isset($form_state['values']['amazons3_secret'])) {
-    $config['credentials'] = new Credentials($form_state['values']['amazons3_key'], $form_state['values']['amazons3_secret']);
+    if (!empty($form_state['values']['amazons3_use_environment_iam'])) {
+      $config['credentials'] = Credentials::factory(array());
```

Comment:
I think factory() has array() as a default parameter, so I think we can remove the argument?